Gems/ScriptEvents/Code/Include/ScriptEvents/ScriptEventTypes.h

```diff
@@ -10,6 +10,18 @@
 
 #include <ScriptEvents/Internal/VersionedProperty.h>
 
+namespace AZ
+{
+    class BehaviorClass;
+    class BehaviorMethod;
+    namespace Script
+    {
+        namespace Attributes
+        {
+            enum class OperatorType : int;
+        }
+    }
```

Comment:
Also FYI this can use the C++17 [nested namespace](https://en.cppreference.com/w/cpp/language/namespace) declaration

```
namespace AZ::Script::Attributes
{
    enum class OperatorType: int;
}

Reply:
100% agreed, but I really wanted to prevent PR-creep :D

Reply:
The second part of this PR is coming: windows unit/non-unity build fixes plus another 13% speedup for a 23% overall reduction on clang/Linux profile build.